src/libcommonserver/log/log.cpp

```diff
@@ -109,4 +115,393 @@ Log::Log(const log4cplus::tstring &filePath) {
+            continue;
+        }
+
+        if (!IoHelper::copyFileOrDirectory(entry.path(), outputPath / entry.path().filename(), ioError)) {
+            LOG_WARN(Log::instance()->getLogger(),
+                     "Error in IoHelper::copyFileOrDirectory: " << Utility::formatIoError(entry.path(), ioError).c_str());
+            if (ioError == IoErrorDiskFull) {
+                exitCause = ExitCauseNotEnoughDiskSpace;
+                return ExitCodeSystemError;
+            } else {
+                exitCause = ExitCauseUnknown;
+                return ExitCodeSystemError;
+            }
+        }
+    }
+
+    if (!endOfDirectory) {
+        LOG_WARN(Log::instance()->getLogger(),
+                 "Error in DirectoryIterator: " << Utility::formatIoError(logPath, ioError).c_str());
+        exitCause = ExitCauseUnknown;
+        return ExitCodeSystemError;
+    }
+
+    exitCause = ExitCauseUnknown;
+    return ExitCodeOk;
+}
+
+ExitCode Log::compressLogFiles(const SyncPath &directoryToCompress, ExitCause &exitCause,
+                               std::function<void(int)> progressCallback) {
+    IoHelper::DirectoryIterator dir;
+    IoError ioError = IoErrorUnknown;
+    if (!IoHelper::getDirectoryIterator(directoryToCompress, true, ioError, dir)) {
+        LOG_WARN(Log::instance()->getLogger(),
+                 "Error in DirectoryIterator: " << Utility::formatIoError(directoryToCompress, ioError).c_str());
+        exitCause = ExitCauseUnknown;
+        return ExitCodeSystemError;
+    }
+
+    bool progressMonitoring = progressCallback != nullptr;
+    float nbFiles = 0;
+    DirectoryEntry entry;
+
+    if (progressMonitoring) {
+        progressCallback(0);
+        bool endOfDirectory = false;
+        while (dir.next(entry, endOfDirectory, ioError) && !endOfDirectory) {
+            nbFiles++;
+        }
+        if (!IoHelper::getDirectoryIterator(directoryToCompress, true, ioError, dir)) {
+            LOG_WARN(Log::instance()->getLogger(),
+                     "Error in DirectoryIterator: " << Utility::formatIoError(directoryToCompress, ioError).c_str());
+            exitCause = ExitCauseUnknown;
+            return ExitCodeSystemError;
+        }
+    }
+
+    float progress = 0.0;
+    bool endOfDirectory = false;
+    while (dir.next(entry, endOfDirectory, ioError) && !endOfDirectory) {
+        std::string entryPath = entry.path().string();
+        if (entryPath.find(".gz") != std::string::npos) {
+            continue;
+        }
+
+        ItemType itemType;
+        const bool success = IoHelper::getItemType(entryPath, itemType);
+        ioError = itemType.ioError;
+        if (!success) {
+            exitCause = ExitCauseUnknown;
+            return ExitCodeSystemError;
+        }
+
+        if (itemType.nodeType != NodeTypeFile) {
+            continue;
+        }
+
+        QString destPath = QString::fromStdString(entryPath + ".gz");
```

Comment:
Is it possible to not use `Qt` functions but only `std` functions?

Reply:
I created a ticket: https://infomaniak.atlassian.net/browse/KDESKTOP-845